service/app/controllers/TelemetryController.java

```diff
@@ -0,0 +1,80 @@
+package controllers;
+
+import com.google.common.net.HttpHeaders;
+import org.apache.commons.lang3.StringUtils;
+import org.sunbird.common.exception.ProjectCommonException;
+import org.sunbird.common.models.util.LoggerEnum;
+import org.sunbird.common.models.util.ProjectLogger;
+import org.sunbird.common.request.Request;
+import org.sunbird.common.responsecode.ResponseCode;
+import org.sunbird.controllers.BaseController;
+import play.libs.F;
+import play.libs.F.Promise;
+import play.libs.Json;
+import play.mvc.Http.RawBuffer;
+import play.mvc.Http.RequestBody;
+import play.mvc.Result;
+import util.Constant;
+import util.Message;
+
+/**
+ * Telemetry controller handles Telemetry APIs.
+ *
+ * @author Mahesh Kumar Gangula
+ */
+public class TelemetryController extends BaseController {
+  /**
+   * Save telemetry information. Request body either with content type application/json or content
+   * encoding gzip.
+   *
+   * @return Return a promise for telemetry API result.
+   */
+  public F.Promise<Result> save() {
+    try {
+      String contentTypeHeader = request().getHeader(HttpHeaders.CONTENT_TYPE);
+      String encodingHeader = request().getHeader(HttpHeaders.ACCEPT_ENCODING);
+      Request request = new Request();
+      request.setOperation(Constant.DISPATCH_TELEMETRY_OPERATION_NAME);

```

Comment:
Operation and actor names should be coming from an enum...

Reply:
@ActorConfig(
  tasks = {Constant.DISPATCH_TELEMETRY_TO_KAFKA}, // this required Constant String , in case of keeping an enum , we can't directly use it.
  asyncTasks = {}
)